xarray/core/accessor_str.py

```diff
@@ -865,49 +1061,838 @@ def replace(self, pat, repl, n=-1, case=None, flags=0, regex=True):
+        array([[['', '', 'abc', 'def'],
+                ['', 'spam', 'eggs', 'swallow'],
+                ['', '', '', 'red_blue']],
+        <BLANKLINE>
+               [['test0', 'test1', 'test2', 'test3'],
+                ['', '', '', ''],
+                ['abra', 'ka', 'da', 'bra']]], dtype='<U8')
+        Dimensions without coordinates: X, Y, ZZ
+
+        Split once and put the results in lists
+
+        >>> values.str.rsplit(dim=None, maxsplit=1)
+        <xarray.DataArray (X: 2, Y: 3)>
+        array([[list(['abc', 'def']), list(['spam\\t\\teggs', 'swallow']),
+                list(['red_blue'])],
+               [list(['test0\\ntest1\\ntest2', 'test3']), list([]),
+                list(['abra  ka\\nda', 'bra'])]], dtype=object)
+        Dimensions without coordinates: X, Y
+
+        Split as many times as needed and put the results in a list
+
+        >>> values.str.rsplit(dim=None)
+        <xarray.DataArray (X: 2, Y: 3)>
+        array([[list(['abc', 'def']), list(['spam', 'eggs', 'swallow']),
+                list(['red_blue'])],
+               [list(['test0', 'test1', 'test2', 'test3']), list([]),
+                list(['abra', 'ka', 'da', 'bra'])]], dtype=object)
+        Dimensions without coordinates: X, Y
+
+        Split only on spaces
+
+        >>> values.str.rsplit(dim="ZZ", sep=" ")
+        <xarray.DataArray (X: 2, Y: 3, ZZ: 3)>
+        array([[['', 'abc', 'def'],
+                ['', '', 'spam\\t\\teggs\\tswallow'],
+                ['', '', 'red_blue']],
+        <BLANKLINE>
+               [['', '', 'test0\\ntest1\\ntest2\\n\\ntest3'],
+                ['', '', ''],
+                ['abra', '', 'ka\\nda\\tbra']]], dtype='<U24')
+        Dimensions without coordinates: X, Y, ZZ
+
+        See Also
+        --------
+        DataArray.str.split
+        str.rsplit
+        pandas.Series.str.rsplit
+        """
+        if sep is not None:
+            sep = self._obj.dtype.type(sep)
+
+        if dim is None:
+            f = lambda x: x.rsplit(sep, maxsplit)
+            return self._apply(f, dtype=np.object_)
+
+        if not self._obj.size:
+            return self._obj.copy().expand_dims({dim: 0}, -1)
+
+        f_count = lambda x: max(len(x.rsplit(sep, maxsplit)), 1)
+        maxsplit = self._apply(f_count, dtype=np.int_).max().data.tolist() - 1
+
+        def _dorsplit(mystr, sep=sep, maxsplit=maxsplit, dtype=self._obj.dtype):
+            res = mystr.rsplit(sep, maxsplit)
+            if len(res) < maxsplit + 1:
+                res = [""] * (maxsplit + 1 - len(res)) + res
+            return np.array(res, dtype=dtype)
+
+        # dtype MUST be object or strings can be truncated
+        # See: https://github.com/numpy/numpy/issues/8352
+        return self._apply(
+            _dorsplit,
+            dtype=np.object_,
+            output_core_dims=[[dim]],
+            output_sizes={dim: maxsplit},
+        ).astype(self._obj.dtype.kind)
+
+    def get_dummies(
+        self,
+        dim: Hashable,
```

Comment:
```suggestion
        dim: Hashable = "groups",
```
Or `"dummies"`?

Reply:
And what if there is already a dimension named "groups"?  This would be a serious problem if someone does operations multiple times.  

Also, I checked how xarray normally handles automatic dimension names, and except for the constructor it generally doesn't.